utils/kubernetes/kubeutil.py

```diff
@@ -56,27 +60,107 @@ def __init__(self, instance=None):
                 instance = {}
 
         self.method = instance.get('method', KubeUtil.DEFAULT_METHOD)
-        self.host = instance.get("host") or self.docker_util.get_hostname()
-        self.kubelet_host = os.environ.get('KUBERNETES_KUBELET_HOST') or self.host
         self._node_ip = self._node_name = None  # lazy evaluation
         self.host_name = os.environ.get('HOSTNAME')
 
-        self.cadvisor_port = instance.get('port', KubeUtil.DEFAULT_CADVISOR_PORT)
-        self.kubelet_port = instance.get('kubelet_port', KubeUtil.DEFAULT_KUBELET_PORT)
-
-        self.kubelet_api_url = '%s://%s:%d' % (self.method, self.kubelet_host, self.kubelet_port)
-        self.cadvisor_url = '%s://%s:%d' % (self.method, self.kubelet_host, self.cadvisor_port)
+        # apiserver
         self.kubernetes_api_url = 'https://%s/api/v1' % (os.environ.get('KUBERNETES_SERVICE_HOST') or self.DEFAULT_MASTER_NAME)
 
+        # kubelet
+        self.ssl_settings = self._init_ssl_settings(instance)
+        self.kubelet_api_url = self._locate_kubelet(instance)
+        if not self.kubelet_api_url:
+            log.exception("Kubernetes check exiting, cannot run without access to kubelet.")
```

Comment:
Should we raise an exception instead ? You shouldn't use `log.exception` if there is no exception btw.